pulpcore/cli/ansible/remote.py

```diff
@@ -0,0 +1,197 @@
+import click
+
+import yaml
+
+from typing import Any, Dict, Optional, Callable
+
+from pulpcore.cli.common.generic import (
+    list_entities,
+    show_by_name,
+    destroy_by_name,
+)
+from pulpcore.cli.common.context import (
+    PulpContext,
+    pass_pulp_context,
+    pass_entity_context,
+    PulpEntityContext,
+    EntityDefinition,
+)
+from pulpcore.cli.ansible.context import (
+    PulpAnsibleRoleRemoteContext,
+    PulpAnsibleCollectionRemoteContext,
+)
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "remote_type",
+    type=click.Choice(["role", "collection"], case_sensitive=False),
+    default="collection",
+    is_eager=True,
+)
+@pass_pulp_context
+@click.pass_context
+def remote(ctx: click.Context, pulp_ctx: PulpContext, remote_type: str) -> None:
+    if remote_type == "role":
+        ctx.obj = PulpAnsibleRoleRemoteContext(pulp_ctx)
+    elif remote_type == "collection":
+        ctx.obj = PulpAnsibleCollectionRemoteContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+def remote_options(f: Callable[..., Any]) -> Callable[..., Any]:
+    @click.option("--ca-cert", help="a PEM encoded CA certificate")
+    @click.option("--client-cert", help="a PEM encoded client certificate")
+    @click.option("--client-key", help="a PEM encode private key")
+    @click.option("--tls-validation", help="if True, TLS peer validation must be performed")
+    @click.option("--proxy-url", help="format: scheme//user:password@host:port")
+    @click.option("--username", help="used for authentication when syncing")
+    @click.option("--password")
+    @click.option("--download-concurrency", help="total number of simultaneous connections")
+    @click.option("--policy", help="policy to use when downloading")
+    @click.option("--total-timeout", help="aiohttp.ClientTimeout.total for download connections")
+    @click.option(
+        "--connect-timeout", help="aiohttp.ClientTimeout.connect for download connections"
+    )
+    @click.option(
+        "--sock-connect-timeout", help="aiohttp.ClientTimeout.sock_connect for download connections"
+    )
+    @click.option(
+        "--sock-read-timeout", help="aiohttp.ClientTimeout.sock_read for download connections"
+    )
+    def config_options(**kwargs: Dict[str, Any]) -> Any:
+        f(**kwargs)
+
+    return config_options
+
+
+def check_collection_options(
+    remote_ctx: PulpEntityContext,
+    requirements_file: Any,
+    requirements: Optional[str],
+    auth_url: Optional[str],
+    token: Optional[str],
+) -> EntityDefinition:
+
+    body: EntityDefinition = dict()
+    if isinstance(remote_ctx, PulpAnsibleRoleRemoteContext) and any(
+        [requirements_file, requirements, auth_url, token]
```

Comment:
can you add a test where you hit one of those and check for the `"${ERROUTPUT}"`?